keras_nlp/models/bert_tasks.py

```diff
@@ -0,0 +1,176 @@
+# limitations under the License.
+
+"""BERT model task heads and preconfigured versions."""
+
+from tensorflow import keras
+
+from keras_nlp.models.bert import BertBase
+from keras_nlp.models.bert import BertLarge
+from keras_nlp.models.bert import BertMedium
+from keras_nlp.models.bert import BertSmall
+from keras_nlp.models.bert import BertTiny
+from keras_nlp.models.bert import bert_kernel_initializer
+
+# Metadata for loading pretrained model weights.
+backbone_checkpoints = {
+    "tiny_uncased_en": {
+        "model": BertTiny,
+        "weights": "uncased_en",
+        "description": (
+            "Tiny size of BERT where all input is lowercased. "
+            "Trained on English Wikipedia + BooksCorpus."
+        ),
+    },
+    "small_uncased_en": {
+        "model": BertSmall,
+        "weights": "uncased_en",
+        "description": (
+            "Small size of BERT where all input is lowercased. "
+            "Trained on English Wikipedia + BooksCorpus."
+        ),
+    },
+    "medium_uncased_en": {
+        "model": BertMedium,
+        "weights": "uncased_en",
+        "description": (
+            "Medium size of BERT where all input is lowercased. "
+            "Trained on English Wikipedia + BooksCorpus."
+        ),
+    },
+    "base_uncased_en": {
+        "model": BertBase,
+        "weights": "uncased_en",
+        "description": (
+            "Base size of BERT where all input is lowercased. "
+            "Trained on English Wikipedia + BooksCorpus."
+        ),
+    },
+    "base_cased_en": {
+        "model": BertBase,
+        "weights": "cased_en",
+        "description": (
+            "Base size of Bert where case is maintained. "
+            "Trained on English Wikipedia + BooksCorpus."
+        ),
+    },
+    "base_zh": {
+        "model": BertBase,
+        "weights": "zh",
+        "description": ("Base size of BERT. Trained on Chinese Wikipedia."),
+    },
+    "base_multi_cased": {
+        "model": BertBase,
+        "weights": "multi_cased",
+        "description": ("Base size of BERT. Trained on Chinese Wikipedia."),
+    },
+    "large_uncased_en": {
+        "model": BertLarge,
+        "weights": "uncased_en",
+        "description": (
+            "Large size of BERT where all input is lowercased. "
+            "Trained on English Wikipedia + BooksCorpus."
+        ),
+    },
+    "large_cased_en": {
+        "model": BertLarge,
+        "weights": "cased_en",
+        "description": (
+            "Base size of Bert where case is maintained. "
+            "Trained on English Wikipedia + BooksCorpus."
+        ),
+    },
+}
+
+
+class BertClassifier(keras.Model):
+    """BERT encoder model with a classification head.
+
+    Args:
+        backbone: A string, `keras_nlp.models.BertCustom` or derivative such as
+            `keras_nlp.models.BertBase` to encode inputs. If a string, should be
+            one of `keras_nlp.models.bert_tasks.backbone_checkpoints`.
```

Comment:
So this is something a user would print out? I think we might want to think about how we export this (and other such lists for preprocessing and backbone models).

We probably should explicitly choose where to export and document these symbols through `__init__.py` rather than couple our file structure. We also may want to add a `bert` directory, so we can control the exports in `keras_nlp.models.bert` via an `__init__.py`. No need to figure all that out on this PR though.

For now, should we just do the same format string with dict keys approach we did for the other bert symbols?

Reply:
Agree on the `bert/` directory, although I may file an issue and reogranize `models/` in a followup PR.

I was worried about the large number of checkpoints and long names that the formatting would be horrible. Is there any way to prevent the docstring from having one 1000 char line? I of course would have made the checkpoint names a string-backed `enum` and called it a day!

Reply:
Fixed for now by adding all checkpoints to the docstring. For the longer term added a TODO in `bert.py` about using keras.io to programatically document checkpoint coverage.